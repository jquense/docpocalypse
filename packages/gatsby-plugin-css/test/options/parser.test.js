const { webpack } = require('../helpers');

describe('Options', () => {
  test('Parser - {String}', () => {
    const config = {
      rules: [
        {
          test: /\.sss$/,
          use: {
            loader: require.resolve('../../postcss-loader'),
            options: {
              parser: 'sugarss',
            },
          },
        },
      ],
    };

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a {\\n  color: black\\n}\\n"');
      expect(source).toMatchSnapshot();
    });
  });

  test('Parser - {Object}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          ident: 'postcss',
          parser: require('sugarss'),
        },
      },
    };

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a {\\n  color: black\\n}\\n"');
      expect(source).toMatchSnapshot();
    });
  });
});
