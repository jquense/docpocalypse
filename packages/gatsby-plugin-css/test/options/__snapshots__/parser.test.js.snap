// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Parser - {Object} 1`] = `"module.exports = \\"a {\\\\n  color: black\\\\n}\\\\n\\""`;

exports[`Options Parser - {String} 1`] = `"module.exports = \\"a {\\\\n  color: black\\\\n}\\\\n\\""`;
